src/app/providers/route-by-category/route-by-category.ts

```diff
@@ -0,0 +1,101 @@
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { TestCategory } from '@dvsa/mes-test-schema/category-definitions/common/test-category';
+import { getPageNameByCategoryAndKey, PageNameKeys } from '@pages/page-names.constants';
+
+@Injectable()
+export class RouteByCategoryProvider {
+  constructor(
+    public router: Router,
+  ) {
+  }
+
+  async navigateToPage(page: PageNameKeys, category?: TestCategory): Promise<void> {
+    const { config } = this.router;
+    const categoryPage = category ? getPageNameByCategoryAndKey(category, page) : page;
+    const pageAlias = this.categoryToPage(category);
+    const pageName = this.pagePath(page);
+    const importPath = category ? `${pageName}/${pageAlias}/${pageName}.${pageAlias}` : `${pageName}/${pageName}`;
+    config.push({
+      path: categoryPage,
+      loadChildren: () =>
+        import(`../../pages/${importPath}.module`)
+          .then((m) => m[`${categoryPage}Module`]),
+    });
+    await this.router.navigate([categoryPage]);
+  }
+
+  categoryToPage(category: TestCategory): string {
```

Comment:
think this name needs to be more descriptive of what it's doing e.g. `getPageAliasByCategory()` ?

Reply:
function name changed